libs/command/context.go

```diff
@@ -0,0 +1,45 @@
+package command
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
+
+// key is a package-local type to use for context keys.
+//
+// Using an unexported type for context keys prevents key collisions across
+// packages since external packages cannot create values of this type.
+type key int
+
+const (
+	// execIDKey is the context key for the execution ID.
+	// The value of 1 is arbitrary and can be any number.
+	// Other keys in the same package must have different values.
+	execIDKey = key(1)
+)
+
+func SetExecId(ctx context.Context) context.Context {
```

Comment:
Set implies (to me) that the user specifies a value. That's not the case here. Maybe `GenerateExecutionID`?